src/markdown.ts

```diff
@@ -282,7 +282,21 @@ export interface ParseOptions {
   md?: MarkdownIt;
 }
 
-export function mdparser({markdownIt}: {markdownIt?: (md: MarkdownIt) => MarkdownIt} = {}): MarkdownIt {
+function cleanPath(href: string, clean: boolean): string {
+  if (href.startsWith("./") || href.startsWith("../")) {
+    const g = href.match(/(?<dir>[^#?]*\/)((?<file>[^/?#]*?)([.]html)?)(?<q>[?#].*)?$/)?.groups;
```

Comment:
Maybe we should use `URL` to parse, and `dirname`, `basename`, `extname`? Regular expressions…